src/primitives/transaction.cpp

```diff
@@ -209,6 +209,32 @@ bool CTransaction::IsCoinStake() const
     return (vout.size() >= 2 && vout[0].IsEmpty());
 }
 
+bool CTransaction::CheckColdStake() const
+{
+    // tx is a coinstake tx
+    if (!IsCoinStake())
+        return false;
+
+    // all inputs have the same pubKeyScript
+    CScript firstScript = vin[0].scriptSig;
+    if (vin.size() > 1) {
+        for (unsigned int i=1; i<vin.size(); i++)
+            if (vin[i].scriptSig != firstScript) return false;
+    }
+
+    // all outputs except first (coinstake marker) and last (masternode payout)
+    // have the same pubKeyScript
+    firstScript = vout[0].scriptPubKey;
```

Comment:
is it me or this method will always return false?, first output in the coinStake tx is the marker, this need to be `vout[1].scriptPubKey` and checked up until the last mn payout output which could or not be there (MN payout enforcement).

Reply:
`f00e3d9bb9c1d567465dae47f064cc4948b7903c` adds a check against last output when masternode payouts are disabled. The first output, as stated in the comment above, should indeed be skipped (nice catch).
The tests don't fail here because `vout.size() == 2` for coinstakes on regtest.

Reply:
ok, one point less then. Only the firstScript need to be modified there and done.
 I'm checking commit by commit, will get to `f00e3d9b` soon 🚜 .